src/lib.rs

```diff
@@ -208,6 +212,31 @@ impl ChildIndex {
 #[derive(Clone, Copy, Debug, PartialEq)]
 struct ChainCode([u8; 32]);
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct DiversifierIndex([u8; 11]);
+
+impl DiversifierIndex {
+    fn new() -> Self {
+        DiversifierIndex([0; 11])
+    }
+
+    pub fn increment(&mut self) -> Result<(), ()> {
```

Comment:
Not sure I understand this code. You have a number represented by 11 u8's and you're trying to implement a +1 function?

Reply:
Right that seems fine